spring-cloud-dataflow-docs/src/main/asciidoc/task-developer-guide.adoc

```diff
@@ -0,0 +1,355 @@
++
+Now we can see the Exception we added coming through in the output, as an Exception with a stack trace.
+. Remove or comment out the Exception that we throw on line 26 (so that the next lessons work correctly).
+
+[[task-dev-guide-building-timestamp-task-adding-pre-post-processing]]
+=== Adding Pre- and Post-processing
+
+Spring Cloud Task includes the ability to run additional processing both before and after the task. To add both features to our current sample application:
+
+. Open `src/main/java/io/spring/TasklabApplication.java` in either an IDE or a text editor.
+. Copy the following code below line 30:
++
+[source,java]
+----
+@BeforeTask
+public void beforeTask(TaskExecution taskExecution) {
+  System. out.println("Before TASK");
+}
+
+@AfterTask
+public void afterTask(TaskExecution taskExecution) {
+  System. out.println("After TASK");
+}
+----
+. From the command line, run `mvn clean package`.
+. From the command line, run `java -jar target/tasklab-0.0.1-SNAPSHOT.jar`.
++
+Now the output includes lines that print both `BEFORE TASK` and `AFTER TASK`.
+
+[[task-dev-guide-building-timestamp-task-bonus-adding-mysql]]
+=== Bonus Step: Adding a MySQL Database
+
+Nearly always, a real-world Spring Cloud Task needs to use a persistent (rather than an in-memory) database.
+In this example, we show how to add a MySQL database (MariaDB) to our Task.
+To do so:
+
+. Open the `pom.xml` file.
+. On line 46, add the following dependency:
++
+[source,xml]
+----
+<dependency>
+  <groupId>org.mariadb.jdbc</groupId>
+  <artifactId>mariadb-java-client</artifactId>
+</dependency>
+----
+. From the command line, run `mvn clean package`.
+. From the command line, run `java -jar target/tasklab-0.0.1-SNAPSHOT.jar`.
++
+If you examine the contents of your database, you should now see the task in the `TASK_EXECUTION` table.
+
+== Adding Spring Cloud Task to a Spring Batch Application
+
+NOTE: This project expects that you have completed the <<task-dev-guide-building-timestamp-task,previous project>>.
+
+You can use Spring Cloud Task within a Spring Batch application.
+In fact, that was a key goal of Spring Could Task.
+Because a Task has a finite duration (that is, it is not a continuing process), it is a natural fit for Spring Batch, which naturally deals with processes that have finite durations.
+
+This guide walks through integrating Spring Cloud Task with Spring Batch in a sample application.
+It consists of the following procedures:
+
+. <<task-dev-guide-building-batch-task-setting-up-environment>>
+. <<task-dev-guide-building-batch-task-creating-first-task>>
+. <<task-dev-guide-building-batch-task-creating-first-batch-task>>
+
+[[task-dev-guide-building-batch-task-setting-up-environment]]
+=== Setting up the Environment
+
+Before we can do anything else, we have to set up our work environment. First, we must install RabbitMQ, as described https://github.com/cppwfs/DNDataflow/blob/master/labs/InstallRabbitnMQ.pdf[in this document].
```

Comment:
Should the sample be moved to SCDF samples so we can link to that repo instead?

Reply:
It's not really a sample. It's really a separate set of instructions. That aside, it ought to be in the same repository as the other content. My larger concern is whether we should have it at all. I suspect there are installation instructions for RabbitMQ in RabbitMQ's official documentation. We should link to those rather than maintain our own, unless we need to do something specific. Even then, our instructions should start with the official instructions and add whatever we need.

I'll look into it.

Reply:
+1 to not include Rabbit setup instructions in "SCDF docs". Let's remove that and add a note about messaging middleware as a prerequisite.

Reply:
My question is that for this demo do we even need Rabbit MQ instructions.   While rabbit is required for some of the features offered by Task, they are not present in this tutorial.

Reply:
I agree. I should have made that (lack of) connection yesterday. I have removed it. Thanks to both of you.